assets/blueprints/genesis_helper/src/lib.rs

```diff
@@ -0,0 +1,236 @@
+                scrypto_encode(&EpochManagerCreateInput {
+                    validator_owner_token,
+                    component_address: epoch_manager_component_address,
+                    validator_set: validators_with_initial_stake,
+                    initial_epoch,
+                    rounds_per_epoch,
+                    num_unstake_epochs,
+                })
+                .unwrap(),
+            );
+
+            // ...and distribute the LP tokens to stakers
+            for (validator_idx, mut lp_bucket) in lp_buckets.into_iter().enumerate() {
+                let stakes = genesis_data.stakes.remove(&validator_idx).unwrap_or(vec![]);
+                for (account_idx, stake_xrd_amount) in stakes {
+                    // TODO: currently xrd amount matches stake tokens amount, but can this change later on?
+                    let stake_bucket = lp_bucket.take(stake_xrd_amount);
+                    let account_address = genesis_data.accounts[account_idx];
+                    let _: () = Runtime::call_method(
+                        account_address,
+                        "deposit",
+                        scrypto_encode(&AccountDepositInput {
+                            bucket: stake_bucket,
+                        })
+                        .unwrap(),
+                    );
+                }
+                lp_bucket.drop_empty();
+            }
+
+            // Allocate XRD
+            for (account_idx, xrd_amount) in genesis_data.xrd_balances.into_iter() {
+                let account_address = genesis_data.accounts[account_idx];
+                let bucket = whole_lotta_xrd.take(xrd_amount);
+                let _: () = Runtime::call_method(
+                    account_address,
+                    "deposit",
+                    scrypto_encode(&AccountDepositInput { bucket }).unwrap(),
+                );
+            }
+
+            // return the remainder
+            whole_lotta_xrd
+        }
+
+        fn create_resource(
+            resource: GenesisResource,
+            initial_supply: Decimal,
+            initial_allocation: BTreeMap<ComponentAddress, Decimal>,
+            owner_with_mint_and_burn_rights: Option<ComponentAddress>,
+        ) -> () {
+            let resource_address = ResourceAddress::Fungible(resource.address_bytes.clone());
+
+            // Just a sanity check that XRD wasn't acccidentally included in genesis resources
+            if resource.symbol.eq_ignore_ascii_case("XRD") {
+                panic!("XRD shouldn't be included in genesis resources");
+            }
+
+            let mut metadata = BTreeMap::new();
+            metadata.insert("symbol".to_owned(), resource.symbol.clone());
+            metadata.insert("name".to_owned(), resource.name);
+            metadata.insert("description".to_owned(), resource.description);
+            metadata.insert("url".to_owned(), resource.url);
+            metadata.insert("icon_url".to_owned(), resource.icon_url);
```

Comment:
Can we add a TODO to update this to use the metadata Url value once that is implemented?